### PR TITLE
Issue 63

### LINKED ISSUE #63: Split percentages cannot exceed 100% of the transaction - allow to specify max allowed overshoot
this is related to #62 

After adding the split sum percentage to the err message i get:
```
pydantic.error_wrappers.ValidationError: 1 validation error for Transaction
splits
  Split percentages cannot exceed 100% of the transaction: 100.02 (type=value_error)
```

it would be good to be specify the allowed overshoot. Currently it's 0.0.1

### PR DESCRIPTION
**Pull request by WolfgangFahl**
fixes #63

### CODE REVIEW

**Comment by WolfgangFahl** [comment]
The test might have to reset the split_overshoot after being done

File: tests/test_transaction.py [hunk 1]
```diff
@@ -76,6 +76,22 @@ def test_create_transaction_with_splits_percent_too_high():
         )
 
 
+def test_split_overshoot_issue63():
```
Comment (isaacharrisholt) [inline tests/test_transaction.py:79]: As you might see with other tests, the issue is typically referenced in the docstring, not the function name

File: quiffen/core/transaction.py [hunk 2]
```diff
@@ -21,6 +21,16 @@
...
 
+class TransactionConfig:
+    """
+    transaction configuration settings
+    """
+
+    # maximum percentage overshoot of a split transaction above
+    # 100%
+    max_split_overshoot: float = 0.01
+
```
Comment (isaacharrisholt) [inline quiffen/core/transaction.py:32]: I'm not sure I'm a fan of storing config on a class like this. Also, if this is intended to be user-editable, then perhaps there's a better way of approaching this than a global config object.

**Review by isaacharrisholt — CHANGES_REQUESTED** [review]
When introducing new functionality like this, it's generally good GitHub etiquette to come to a solution in an issue before implementing the code. Perhaps it would be good to use issue #63 for that purpose here.

**Comment by WolfgangFahl** [comment]
Still this is an improvement and doesn't break anything. I don't see why this shouldn't be accepted ASAP.

**Comment by isaacharrisholt** [comment]
> Still this is an improvement and doesn't break anything. I don't see why this shouldn't be accepted ASAP.

Perhaps, but I also have to consider the longer-term view for this project. I'm ultimately going to be the one maintaining this, and if I wanted to change it, which I likely would, I'd have to release a new major version of Quiffen. This needs more discussion.

**Comment by isaacharrisholt** [comment]
Closing due to inactivity. Feel free to open a discussion thread to discuss implementation.

**Pull request closed without merging**